src/commands/deleteVirtualMachine/getResourcesAssociatedToVm.ts

```diff
@@ -0,0 +1,90 @@
+            associatedResources.push({
+                resourceName: networkName, resourceType: networkInterfaceLabel,
+                deleteMethod: async (): Promise<void> => { await networkClient.networkInterfaces.deleteMethod(resourceGroupName, networkName); }
+            });
+
+            if (networkInterface.ipConfigurations) {
+                for (const ipConfigurations of networkInterface.ipConfigurations) {
+                    if (ipConfigurations.publicIPAddress?.id) {
+                        const publicIpName: string = getNameFromId(ipConfigurations.publicIPAddress.id);
+                        associatedResources.push({
+                            resourceName: publicIpName, resourceType: localize('publicip', 'public IP address'),
+                            deleteMethod: async (): Promise<void> => { await networkClient.publicIPAddresses.deleteMethod(resourceGroupName, publicIpName); }
+                        });
+
+                    }
+
+                    if (ipConfigurations.subnet?.id) {
+                        const subnetId: string = ipConfigurations.subnet.id;
+                        // example of subnet id: '/subscriptions/9b5c7ccb-9857-4307-843b-8875e83f65e9/resourceGroups/linux-vm/providers/Microsoft.Network/virtualNetworks/linux-vm/subnets/default'
+                        const virtualNetworkName: string = subnetId.split('/')[8];
+                        associatedResources.push({
+                            resourceName: virtualNetworkName, resourceType: virtualNetworkLabel,
+                            deleteMethod: async (): Promise<void> => { await networkClient.virtualNetworks.deleteMethod(resourceGroupName, virtualNetworkName); }
+                        });
+
+                        const subnetName: string = getNameFromId(subnetId);
+                        try {
+                            const subnet: NetworkManagementModels.Subnet = await networkClient.subnets.get(resourceGroupName, virtualNetworkName, subnetName);
+                            if (subnet.networkSecurityGroup?.id) {
+                                const nsgName: string = getNameFromId(subnet.networkSecurityGroup.id);
+                                associatedResources.push({
+                                    resourceName: nsgName, resourceType: localize('networkSecurityGroup', 'network security group'),
+                                    deleteMethod: async (): Promise<void> => { await networkClient.networkSecurityGroups.deleteMethod(resourceGroupName, nsgName); }
+                                });
+                            }
+                        } catch (err) {
+                            // ignore error
+                        }
+                    }
+
+                }
+            }
+        } catch (err) {
+            // ignore error
+        }
+    }
+
+    // if we can't retrieve the disk name, it's highly likely that it's the same as the vmName if it was created from the extension
```

Comment:
How common is this? Did you actually run into it? I feel like I'd err on the side of not including something in the quick pick if it's an unlikely/weird situation